### PR TITLE
Issue 5946: Speedup controller Unit tests

### LINKED ISSUE #5946: Speed up Controller unit tests
**Describe the bug**
Some of the controller unit tests take more than 30s each.

**Problem location**
io.pravega.controller

**Suggestions for an improvement**
Use @BeforeClass annotation for setting up zookeeper and curator client frameworks which would speed up run time of tests.

### PR DESCRIPTION
**Pull request by kotlasaicharanreddy**
Signed-off-by: SaiCharan <kotlasaicharan@yahoo.com>

**Change log description**  
Speed up Controller unit tests

**Purpose of the change**  
Fixes #5946 

**What the code does**  
With theses changes `zookeeper server` and `Curator Client` are setup only once for all tests of a class. Controller unit tests executes in `4m20s`

**How to verify it**  
All the existing tests should continue too pass.

### CODE REVIEW

**Review by shiveshr — APPROVED** [review]
LGTM, @shrids could you please review it too.

File: controller/src/test/java/io/pravega/controller/fault/ControllerClusterListenerTest.java [hunk 1]
```diff
@@ -73,11 +72,13 @@
  */
 @Slf4j
 public class ControllerClusterListenerTest {
+
+    private static final RetryPolicy RETRY_POLICY = new ExponentialBackoffRetry(200, 10, 5000);
```
Comment (shrids) [inline controller/src/test/java/io/pravega/controller/fault/ControllerClusterListenerTest.java:76]: This configuration is the same for all the classes, perhaps we can have the default retry value configured within the PravegaZkCuratorResource.

Comment (kotlasaicharanreddy) [reply controller/src/test/java/io/pravega/controller/fault/ControllerClusterListenerTest.java:76]: Initially I configured it in PravegaZkCuratorResource , but there are some classes where a different RetryPolicy is being used.

For example: 

https://github.com/pravega/pravega/pull/5988/files#diff-cfab5bd3e94538550da7fda0f698ab12eed5cc90c9d0d7b77c2ce73c7f957480R94

File: controller/src/test/java/io/pravega/controller/fault/ControllerClusterListenerTest.java [hunk 2]
```diff
@@ -73,11 +72,13 @@
...
+    @ClassRule
+    public static final PravegaZkCuratorResource PRAVEGA_ZK_CURATOR_RESOURCE = new PravegaZkCuratorResource(RETRY_POLICY);
     @Rule
     public Timeout globalTimeout = new Timeout(30, TimeUnit.HOURS);
```
[the remark this answers is not in the document]
Comment (kotlasaicharanreddy) [reply controller/src/test/java/io/pravega/controller/fault/ControllerClusterListenerTest.java]: Changed it to Seconds. Thanks you.

File: controller/src/test/java/io/pravega/controller/PravegaZkCuratorResource.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+import io.pravega.test.common.TestingServerStarter;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.curator.RetryPolicy;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.CuratorFrameworkFactory;
+import org.apache.curator.test.TestingServer;
+import org.junit.rules.ExternalResource;
+
+
```
[the remark this answers is not in the document]
Comment (kotlasaicharanreddy) [reply controller/src/test/java/io/pravega/controller/PravegaZkCuratorResource.java]: Done.Thanks.

File: controller/src/test/java/io/pravega/controller/PravegaZkCuratorResource.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+            client = CuratorFrameworkFactory.newClient(connectionString, sessionTimeoutMs, connectionTimeoutMs, retryPolicy);
+        }
+        client.start();
+        storeClient = StoreClientFactory.createZKStoreClient(client);
+    }
+
+    @Override
+    @SneakyThrows
+    public void after() {
+        client.close();
```
[the remark this answers is not in the document]
Comment (kotlasaicharanreddy) [reply controller/src/test/java/io/pravega/controller/PravegaZkCuratorResource.java]: Changed accordingly.

**Review by shrids — APPROVED** [review]
LGTM.

**Pull request merged**